pkg/crinit/common/common.go

```diff
@@ -432,15 +433,22 @@ func WaitForPods(cmdOut io.Writer, clientset client.Interface, pods []string, na
 
 func WaitSrvHealthy(cmdOut io.Writer, crClientset client.Interface) error {
 	discoveryClient := crClientset.Discovery()
-	return wait.PollInfinite(podWaitInterval, func() (bool, error) {
+	var innerErr error
+	err := wait.PollImmediate(podWaitInterval, podWaitTimeout, func() (bool, error) {
 		fmt.Fprint(cmdOut, ".")
-		body, err := discoveryClient.RESTClient().Get().AbsPath("/healthz").Do().Raw()
-		if err != nil {
+		body, innerErr := discoveryClient.RESTClient().Get().AbsPath("/healthz").Do().Raw()
+		if innerErr != nil {
 			return false, nil
 		}
 		if strings.EqualFold(string(body), "ok") {
 			return true, nil
 		}
 		return false, nil
 	})
+
+	if err != nil {
+		return innerErr
```

Comment:
I don't think `innerErr` is guaranteed to contain an error, since there's another `false, nil` return inside the condition function. Perhaps we should check if `innerErr` is `nil` before returning and if so, return `err` (the timeout) instead.

Reply:
Ah, good point. I made this change and then simplified the condition a bit.